firefox-ios/Client/Frontend/Home/Homepage Rebuild/Redux/HomepageState.swift

```diff
@@ -23,26 +28,30 @@ struct HomepageState: ScreenState, Equatable {
         self.init(
             windowUUID: homepageState.windowUUID,
             loadInitialData: homepageState.loadInitialData,
-            headerState: homepageState.headerState
+            headerState: homepageState.headerState,
+            navigateTo: homepageState.navigateTo
         )
     }
 
     init(windowUUID: WindowUUID) {
         self.init(
             windowUUID: windowUUID,
             loadInitialData: false,
-            headerState: HeaderState(windowUUID: windowUUID)
+            headerState: HeaderState(windowUUID: windowUUID),
+            navigateTo: nil
         )
     }
 
     private init(
         windowUUID: WindowUUID,
         loadInitialData: Bool,
-        headerState: HeaderState
+        headerState: HeaderState,
+        navigateTo: NavigationDestination?
```

Comment:
for these optional things in the state, that are only set to something specific during a specific action, and nil/false/whatever otherwise, you should have a default initializer, so we don't always have to write them out. Makes our lives easier. I am lazy.

Reply:
haha, will do and embrace my lazy side (if I can find it O_O). `loadInitialData` goes away in my other PR, updated `navigateTo` to be set to `nil`!